test/parallel/test-assert-deep.js

```diff
@@ -1123,3 +1116,33 @@ assert.throws(
   // The descriptor is not compared.
   assertDeepAndStrictEqual(a, { a: 5 });
 }
+
+// Verify object types being identical on both sides.
+{
```

Comment:
these are the only tests that fail without your change

Reply:
Yes, that is correct. The second commit is just there to prevent any further issues like these. It's somewhat independent but it fit together. Would you rather separate the two commits into two PRs?

Reply:
nah this is fine to me, was highlighting here in reference to https://github.com/nodejs/node/pull/30764#discussion_r352772843